File: Elixir/problem_2.ex
# Problem 2: Even Fibonacci numbers
# https://projecteuler.net/problem=2

defmodule Problem2 do
  def fibonacci(1), do: 1
  def fibonacci(2), do: 2
  def fibonacci(n), do: fibonacci(n-1) + fibonacci(n-2)
  def even_sum(n) do
    Enum.filter(fibonacci(1)..fibonacci(n), &(rem(&1, 2) == 0))
      |> Enum.sum
  end
end

# Result:
# > Problem2.fibonacci(5)
# 20

File: r_euler/problem_2.R
# Problem 2: Even Fibonacci numbers
# https://projecteuler.net/problem=2

sum_even_fibonacci <- function(n){
    result <- integer(n)
    result[1] <- 1
    result[2] <- 2
    i <- 3
    while(result[i-1]<n)
        result[i] <- result[i-1] + result[i-2]
        i <- i+1
    sum(result[result %% 2 == 0])
}

# Result:
# > sum_even_fibonacci(4000000)
# Does not end: I have to refactor it
